src/allmydata/util/i2p_provider.py

```diff
@@ -148,13 +154,25 @@ def get_i2p_handler(self):
         launch = self._get_i2p_config("launch", False, boolean=True)
         configdir = self._get_i2p_config("i2p.configdir", None)
         keyfile = self._get_i2p_config("dest.private_key_file", None)
+        options = self._get_i2p_config("i2cp.options", None)
+
+        if options:
+            def transform_escapes(val):
+                return val.replace('\,', '\ESCAPED-COMMA').replace('\:', '\ESCAPED-COLON')
+
+            def unescape(val):
+                return val.replace('\ESCAPED-COMMA', ',').replace('\ESCAPED-COLON', ':')
+
```

Comment:
This all probably deserves a comment, like why is this needed? It was also surprising to me to learn that `'\,'` is the very same as `r'\,'` in this case -- but it might be better to use the `r` (raw) prefixed strings anyway, so the reader knows you meant that to be "backslash comma" not just "comma".